libs/uart/uart.c

```diff
@@ -0,0 +1,67 @@
+/* Atmega 16M1 Uart Library Implementation
+All page numbers refer to
+http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-7647-Automotive-Microcontrollers-ATmega16M1-32M1-64M1-32C1-64C1_datasheet.pdf
+*/
+#include "libs/uart/api.h"
+#include <avr/io.h>
+
+void uart_init(uint16_t baudrate, bool use_interrupt) {
+    // TODO: Implement
+    (void)use_interrupt;
```

Comment:
Can you provide some background on this in a comment? I haven't worked with uart in a while so maybe I'm just rusty but I have no clue when I'd want to set this true or false.